api/main.go

```diff
@@ -55,14 +56,18 @@ func Start() {
 	r.GET("/verification", HandleGetVerificationFlow)
 	r.POST("/verification", HandlePostVerificationFlow)
 
-	r.GET("/get_profile", GetProfile)
-
+	r.POST("/get_profile", middleware.HandleAppAuthorization, HandlePostProfile)
+	r.POST("/verify_app", middleware.HandleAppAuthorization, func(c *gin.Context) {
```

Comment:
Do we need this route?

Reply:
yes, the first route is for the first attempt the the apps that they make directly to nymeria if it throws an error in case where there are no cookie, then the app redirects to eris and eris will the make the request to the second endpoint to verify that the app is authorized to get a redirect or not

Reply:
But what is the requirement of the /verify_app route?

Reply:
the need is mentioned above, this route is used by eris to make sure that in case of redirect only authorized apps will be allowed to make a redirect